### PR TITLE
Check config if provider is systemd

### LINKED ISSUE #16: Centos 7 systemd service does not check config
We need to manually check the config on changes for machines with systemd in order to prevent config errors on restart. I originally thought this was handled by the systemd wrapper script but that is not the case.

### PR DESCRIPTION
**Pull request by armiller**
This will manually check the config on changes and fail within chef if there is an error. Right now I'm checking for the platform version but I'm wondering if there is a way to check if its using the `systemd` provider. Resolves #16 

EDIT: Ran a full kitchen test of each suite and they all passed.

### CODE REVIEW

**Comment by armiller** [comment]
@osuosl-cookbooks/chefs

File: recipes/data_bag.rb [hunk 1]
```diff
@@ -162,3 +162,15 @@
...
+
+execute 'check_haproxy_config' do
+  command "/usr/sbin/haproxy -c -V -f #{conf_file}"
+  action :nothing
+  only_if { platform_family?('rhel') && node['platform_version'] >= '7.0' }
+end
+
+rewind "template[#{conf_file}]" do
+  notifies :run, 'execute[check_haproxy_config]', :immediately
+end
```
Comment (ramereth) [inline recipes/data_bag.rb]: Don't you need to add the following to the recipe?

``` ruby
chef_gem 'chef-rewind'
require 'chef/rewind'
```

Comment (armiller) [inline recipes/data_bag.rb]: ah yes.

Comment (armiller) [inline recipes/data_bag.rb]: @ramereth that is already included in the default recipe.

Comment (ramereth) [inline recipes/data_bag.rb]: Ah, I missed you pulling that in here. Derp, that would explain how this is actually working.

**Comment by ramereth** [comment]
:+1:

**Pull request merged**